tapir/shifts/templates/shifts/user_shifts_overview_tag.html

```diff
@@ -128,5 +128,32 @@ <h5 class="card-header d-flex justify-content-between align-items-center" id="us
                 </a>
             </div>
         </div>
+        <div class="row m-1">
+            <div class="col-4 fw-bold text-end">{% translate "Solidarity" %}:</div>
+            <div class="col-8 d-flex gap-1">
+                {% if user.shift_user_data.get_available_solidarity_shifts and user.shift_user_data.get_account_balance < 0 %}
+                    <form method="post" action="{% url 'shifts:solidarity_shift_used' object.pk%}">
+                        {% csrf_token %}
+                        <button type="submit" class="{% tapir_button_action %} btn-sm"><span class="material-icons">favorite</span>{% translate 'Receive Solidarity' %}</button>
+                    </form>
+                {% else %}
+                    <form method="post" action="{% url 'shifts:solidarity_shift_used' object.pk%}">
+                        {% csrf_token %}
+                        <button type="submit" class="{% tapir_button_action %} btn-sm" disabled><span class="material-icons">favorite</span>{% translate 'Receive Solidarity' %}</button>
```

Comment:
Isn't it more readable the way it is now? In the example, the if condition is short, but with the Receive Solidarity and Give Solidarity Buttons the condition is rather long and, to me, looks more confusing when it is inside the Button code. So, I'd vote for keeping it as it is now. What do you think?

Reply:
Mmmm maybe we should make a tag then? So that we can do something like
```html
<button type="submit" class="{% tapir_button_action %} btn-sm" {% disabled_if_user_cant_receive_solidarity_shift user %}>
```
The tag would just return `"disabled"` or an empty string.

You can search for `@register.simple_tag` to have examples of tags.

Reply:
👍 I created two tags and did it that way